evap/evaluation/models.py

```diff
@@ -470,6 +470,35 @@ class State(models.IntegerChoices):
         verbose_name=_("wait for grade upload before publishing"), default=True
     )
 
+    @property
+    def has_exam_evaluation(self):
+        return self.course.evaluations.filter(name_de="Klausur", name_en="Exam").exists()
+
+    @property
+    def earliest_possible_exam_date(self):
+        return self.vote_start_datetime.date() + timedelta(days=1)
+
+    @transaction.atomic
+    def create_exam_evaluation(self, exam_date: date):
+        self.weight = 9
+        self.vote_end_date = exam_date - timedelta(days=1)
+        self.save()
+        exam_evaluation = Evaluation(
+            course=self.course,
+            name_de="Klausur",
+            name_en="Exam",
+            weight=1,
+            is_rewarded=False,
+            vote_start_datetime=datetime.combine(exam_date + timedelta(days=1), time(8, 0)),
+            vote_end_date=exam_date + timedelta(days=3),
+        )
+        exam_evaluation.save()
+
+        exam_evaluation.participants.set(self.participants.all())
+        for contribution in self.contributions.exclude(contributor=None):
+            exam_evaluation.contributions.create(contributor=contribution.contributor)
```

Comment:
@richardebeling Shouldn't we also copy things like `role`, `textanswer_visibility`, etc. here, or should this just be adjusted by managers afterwards?

Reply:
These values should not be copied; editors have to decide about the permissions here as well.